### PR TITLE
Using Github actions as CI

### PR DESCRIPTION
**Pull request by lekshmideepu**
Introducing Github actions as the CI tool for NEST builds.

### CODE REVIEW

File: extras/gha_build.sh [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+#       Any changes to message numbers (MSGBLDnnnn) or the variable name
+#      'file_names' have effects on the build/test-log parsing process.
+
+
+# Exit shell if any subcommand or pipline returns a non-zero status.
+set -ex
+
+env
+if [ "$xNEST_BUILD_COMPILER" = "CLANG" ]; then
+    export CC=clang-7
```
Comment (heplesser) [inline extras/gha_build.sh:36]: clang-7 seems rather old, the current version is 11. Is there a specific reason to use clang-7?

**Comment by heplesser** [comment]
@lekshmideepu Thanks a lot for this! I am just wondering about one thing (probably something I have not quite understood yet ...): Why do I only see the Travis checks running, but no Github actions? Where would they appear?

**Comment by lekshmideepu** [comment]
> @lekshmideepu Thanks a lot for this! I am just wondering about one thing (probably something I have not quite understood yet ...): Why do I only see the Travis checks running, but no Github actions? Where would they appear?

I was also wondering the same. Ideally it should run under Actions of nest-simulator. I could see that it runs on my repo and not here. Not sure why it is happening. Strange. I am trying to figure this out.

File: extras/gha_build.sh (hunk 1 above)
Comment (lekshmideepu) [reply extras/gha_build.sh]: No specific reason. Thanks for pointing this out. I just added a clang-11 build.

**Comment by lekshmideepu** [comment]
> @lekshmideepu Thanks a lot for this! I am just wondering about one thing (probably something I have not quite understood yet ...): Why do I only see the Travis checks running, but no Github actions? Where would they appear?

I figured this out. I forgot to add the keyword "on pull request". Sorry.  It is now working as expected :-)

**Comment by heplesser** [comment]
@lekshmideepu Great! I have just three suggestions concerning the build matrix before we merge this: 
- We should return to a full build on macOS again (without MUSIC, I think it is still difficult to install on macOS).
- To keep resource use within bounds, I would drop the macOS gcc build.
- I would also consider to drop the clang build on Ubuntu, given that we test against clang on macOS. That is unless there were specific reasons for introducing this combination. 
The last two suggestions are just meant to keep the CO2-footprint of NEST development bounded ;).

**Comment by lekshmideepu** [comment]
Valid suggestions. I agree.

**Comment by heplesser** [comment]
@lekshmideepu I see that the macOS full build fails due to OpenMP problems. Fixing this is a separate issue, so I would suggest that we for this PR stick to the macOS Minimal build after all and create a follow-up for re-instating a full macOS case.

**Comment by lekshmideepu** [comment]
Ok. I think that would be the best solution. Will revert it back.

**Review by heplesser — APPROVED** [review]

File: extras/gha_build.sh [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+    CONFIGURE_MPI="-Dwith-mpi=ON"
+else
+    CONFIGURE_MPI="-Dwith-mpi=OFF"
+fi
+
+if [ "$xPYTHON" = "1" ] ; then
+    export PYTHON_INCLUDE_DIR=`python3 -c "import sysconfig; print(sysconfig.get_path('include'))"`
+    export PYLIB_BASE=lib`basename $PYTHON_INCLUDE_DIR`
+    export PYLIB_DIR=$(dirname `sed 's/include/lib/' <<< $PYTHON_INCLUDE_DIR`)
+    export PYTHON_LIBRARY=`find $PYLIB_DIR \( -name $PYLIB_BASE.so -o -name $PYLIB_BASE.dylib \) -print -quit`
```
Comment (jougs) [inline extras/gha_build.sh:230]: I'm impressed to see that our detection mechanism is even stable enough to work on a completely different CI system!

:smile_cat:

**Review by jougs — APPROVED** [review]
Although I think that our internal variables (e.g. `xNEST_BUILD_TYPE`) would be fine without the leading `x` and others (e.g. `xGSL`) should better be called `NEST_BUILD_FLAG_...` I think we should merge this quickly and improve on such things later on.

In this sense: full approval and again many, many thanks for your hard work and the uncounted iterations this one took you.

File: extras/gha_build.sh (hunk 2 above)
Comment (lekshmideepu) [reply extras/gha_build.sh:230]: > Although I think that our internal variables (e.g. `xNEST_BUILD_TYPE`) would be fine without the leading `x` and others (e.g. `xGSL`) should better be called `NEST_BUILD_FLAG_...` I think we should merge this quickly and improve on such things later on.
> 
> In this sense: full approval and again many, many thanks for your hard work and the uncounted iterations this one took you.

Thank you very much! :-)

**Pull request merged**